tsl/src/compression/compression.c

```diff
@@ -1512,3 +1516,155 @@ update_compressed_chunk_relstats(Oid uncompressed_relid, Oid compressed_relid)
 		CommandCounterIncrement();
 	}
 }
+
+typedef struct CompressSingleRowState
+{
+	Relation in_rel;
+	Relation out_rel;
+	RowCompressor row_compressor;
+	TupleTableSlot *out_slot;
+} CompressSingleRowState;
+
+static TupleTableSlot *compress_singlerow(CompressSingleRowState *cr, TupleTableSlot *in_slot);
+
+CompressSingleRowState *
+compress_row_init(int srcht_id, Relation in_rel, Relation out_rel)
```

Comment:
It's not obvious to me how we can get a certain tuple from e.g. `in_rel` and how this function is being used with regard to the row in question. Maybe a comment would help that explains what the contract is for getting tuples from `Relation` pointers.

Reply:
The second PR should make this clearer: the general pattern is to call xxxInit, xxxExec and then xxxEnd